channeldb/channel.go

```diff
@@ -2676,7 +2727,32 @@ func fetchChanInfo(chanBucket *bbolt.Bucket, channel *OpenChannel) error {
 
 	channel.Packager = NewChannelPackager(channel.ShortChannelID)
 
-	return nil
+	// Finally, read the optional shutdown scripts.
+	if err := readOptionalShutdownScript(
+		chanBucket, localUpfrontShutdownKey, &channel.LocalShutdownScript,
+	); err != nil {
+		return err
+	}
+
+	return readOptionalShutdownScript(
+		chanBucket, remoteUpfrontShutdownKey, &channel.RemoteShutdownScript,
+	)
+}
+
+// readOptionalShutdownScript reads the shutdown script stored under the key
+// provided if it is present. Upfront shutdown scripts are optional, so the
+// function returns with no error if the key is not present.
+func readOptionalShutdownScript(chanBucket *bbolt.Bucket, key []byte,
```

Comment:
naming suggestion: `get....`

Also keep put and get methods close in the file?